charmcraft/application/commands/store.py

```diff
@@ -830,6 +833,167 @@ def run(self, parsed_args):
         emit.message(msg.format(*args))
 
 
+class PromoteCommand(CharmcraftCommand):
+    """Promote a charm in the Store."""
+
+    name = "promote"
+    help_msg = "Promote a charm from one channel to another on Charmhub."
+    overview = textwrap.dedent(
+        """Promote a charm from one channel to another on Charmhub.
+
+        Promotes the current revisions of a charm in a specific channel, as well as
+        their related resources, to another channel.
+
+        The most common use is to promote a charm to a more stable risk value on a
+        single track:
```

Comment:
```suggestion
        """
        Promote a charm from one channel to another on Charmhub.

        Promotes the current revisions of a charm in a specific channel, as well as
        their related resources, to another channel.

        The most common use is to promote a charm to a more stable risk value on a
        single track:
```

I noticed that during the demo and I can see it in documentation as well, I think to make `dedent` work correctly you have to have same indentation in all lines.
It's a bit weirdly rendered here right now
https://canonical-charmcraft--2082.com.readthedocs.build/en/2082/reference/commands/promote/

Reply:
Good catch, thanks. Dedent doesn't require the same indentation everywhere, but it does use the first line's preceding whitespace to determine what to remove. That means the actual correct way is:

```suggestion
        """\
        Promote a charm from one channel to another on Charmhub.

        Promotes the current revisions of a charm in a specific channel, as well as
        their related resources, to another channel.

        The most common use is to promote a charm to a more stable risk value on a
        single track:
```